pyproject.toml

```diff
@@ -1,41 +1,51 @@
-[project]
+[tool.poetry]
 name = "ngsPETSc"
-version = "0.0.5"
-description = "NGSolve/Netgen interface to PETSc."
-readme = "README.md"
+version = "0.0.7a7"
+description = "NGSolve/Netgen interface to PETSc"
 authors = [
-  {name = "Umberto Zerbinati", email = "umberto.zerbinati@maths.ox.ac.uk"},
-  {name = "Patrick E. Farrell", email = "patrick.farrell@maths.ox.ac.uk"},
-  {name = "Stefano Zampini", email = "stefano.zampini@kaust.edu.sa"},
-  {name = "Jack Betteridge", email = "J.Betteridge@imperial.ac.uk"},
+  "Umberto Zerbinati <umberto.zerbinati@maths.ox.ac.uk>",
+  "Patrick E. Farrell <patrick.farrell@maths.ox.ac.uk>",
+  "Stefano Zampini <stefano.zampini@kaust.edu.sa>",
+  "Jack Betteridge <J.Betteridge@imperial.ac.uk>",
 ]
 maintainers = [
-  {name = "Umberto Zerbinati", email = "umberto.zerbinati@maths.ox.ac.uk"},
+  "Umberto Zerbinati <umberto.zerbinati@maths.ox.ac.uk>",
 ]
-license = {file = "LICENSE.txt"}
-dependencies = [
-  "mpi4py",
-  "numpy",
-  "scipy",
+readme = "README.md"
+license = "MIT"
+packages = [
+    {include = "ngsPETSc"}
 ]
-requires-python = ">=3.8"
+documentation = "https://ngspetsc.readthedocs.io/en/latest/"
+repository = "https://github.com/NGSolve/ngsPETSc"
 classifiers = [
   "Programming Language :: Python :: 3",
   "License :: OSI Approved :: MIT License",
   "Operating System :: OS Independent",
   "Development Status :: 3 - Alpha",
 ]
 
-[project.urls]
-Documentation = "https://ngspetsc.readthedocs.io/en/latest/"
-Repository = "https://github.com/NGSolve/ngsPETSc"
+[tool.poetry.dependencies]
+python = "^3.9"
+netgen-mesher = "^6.2"
+netgen-occt = "^7.8"
+petsc4py =  "^3.22.1"
+mpi4py = "^4"
+numpy = "^2"
+scipy = "^1"
```

Comment:
```
$ git grep scipy
Dockerfile:RUN pip install numpy scipy cython mpi4py pytest pytest-mpi
docs/requirements.txt:scipy==1.11.1
docs/src/PETScBasic/pinvt.py.rst:   from scipy.linalg import eigh
docs/src/PETScPC/stokes.py.rst:   from scipy.sparse import coo_matrix
ngsPETSc/utils/firedrake/meshes.py:from scipy.spatial.distance import cdist
pyproject.toml:scipy = "^1"
```

Can we get rid of the scipy dependency `cdist` ?

Reply:
Sure, although it's probably it's own PR. I introduced it as optional but was told it was fine to have a hard dependency so removed the alternative.

See: https://github.com/NGSolve/ngsPETSc/pull/48#discussion_r1787549461

Reply:
We didn't have this dependence a while ago, after @JDBetteridge refactored ngsPETSc curve mesh support for Firedrake he opted to use `cdist` to make the code faster. Maybe we can substitute `cdist` with a function pure python function and then check if the user has installed `scipy` (try: import scipy ...) , if so replace our own `cdist` with scipy `cdist` ?  I think this was the original plan and it is my fault since I suggested @JDBetteridge to make scipy a hard dependence.

Reply:
I'm not sure having scipy as a dependency is the end of the world, it's an easy to install well supported project and comes by default on some Python distributions.